aten/src/ATen/native/cpu/ScatterGatherKernel.cpp

```diff
@@ -193,22 +187,10 @@ struct cpu_scatter_gather_base_kernel {
   }
 
   template <typename func_t>
-  void operator()(Tensor& self, int64_t dim,
+  void operator()(const Tensor& self, int64_t dim,
     const Tensor& index, const Tensor& src,
     const std::string& method_name, func_t& kernel_func) {
 
-    // no-op if index is empty
-    if (index.numel() == 0) {
-      return;
-    }
-
-    dim = maybe_wrap_dim(dim, self.dim());
-
-    scatter_gather_dtype_check(method_name, self, index, src);
-    if (!is_scatter_like) {
-      gather_shape_check(self, dim, index, src);
-    }
-
```

Comment:
The deletions here look suspicious, because the kernel here is used to implement BOTH scatter and gather, but you are only porting gather to structured, so you may have just lost a bunch of dtype checks for scatter. Maybe the easiest way to resolve this is also port scatter to structured, so the deletions are now valid.

Reply:
As far as I understand, `scatter` is already ported, and we do these same checks in the META function. If I remember correctly, the only reason those checks were still there was because of the `gather` kernel.

(`scatter_meta_impl` is called by the META functions)
https://github.com/pytorch/pytorch/blob/839eaa2e91556ecd4532596b4fef18a1c3f6e1c1/aten/src/ATen/native/TensorAdvancedIndexing.cpp#L86-L97

Reply:
Also, something similar to this was discussed here: https://github.com/pytorch/pytorch/pull/57015/files#r636843373. (PR that ported `scatter` to structured kernel)